File: momose/src/models/simpleModelStructures/SimpleModelParser.java
package models.simpleModelStructures;

import models.ModelParser;

import org.xml.sax.Attributes;


public class SimpleModelParser extends ModelParser
 {
  protected static int PAUSETIME=5;
  protected static int VMIN=6;
  protected static int VMAX=7;
	
  private float pauseTime;	
  private float vMin;
  private float vMax;
      	
  public SimpleModelParser()
    {
	 super();
	 pauseTime=0;
	 vMin=0;
	 vMax=5;
    }//Fine costruttore
  
  public float getVMax()
   {
	if(vMax<0)  
	 return 0;
	else
     return vMax;		
   }//Fine getVMax
  
  public float getVMin()
   {
	if(vMin<0)  
	 return 0;
	else
     return vMin;		
   }//Fine getVMax
  
  public float getPauseTime()
   {
	if(pauseTime<0)  
	 return 0;
	else
     return pauseTime;		
   }//Fine getInterval  
  
  public void startElement (String uri, String name,String qName, Attributes atts)
   {
    //Richiamo il metodo del padre (controlla numero nodi,raggio nodi e raggio antenna)	  
    super.startElement(uri,name,qName,atts);	  
	
    //Intervallo di tempo <interval>value<interval>
    if(qName.equals("pauseTime")) 
	 { actTag=PAUSETIME; }
    
    //Velocita' minima <vMin>value</vMin>
    if(qName.equals("vMin")) 
	 { actTag=VMIN; }

    //Velocita' massima <vMax>value</vMax>
    if(qName.equals("vMax")) 
	 { actTag=VMAX; }
   }//Fine startElement
 
  public void characters(char[] ch,int start,int length)
   {
	//Richiamo il metodo del padre  
	super.characters(ch,start,length);  
	  
    String str=new String(ch,start,length);
    //Intervallo
    if(actTag==PAUSETIME) 
	 { 
      pauseTime=Float.valueOf(str);
      actTag=-1;
     }

    //Velocita' minima
    if(actTag==VMIN) 
	 {
      vMin=Float.valueOf(str);
      actTag=-1;
     }
   
    //Velocità massima
    if(actTag==VMAX) 
	 { 
      vMax=Float.valueOf(str);
      actTag=-1;
     }
   }//Fine characters
}//Fine SimpleModelParser

File: momosecpp/src/SimpleModelParser.cpp
#include "SimpleModelParser.h"


SimpleModelParser::SimpleModelParser():ModelParser()
  { 
    pauseTime=0;
    vMin=0;
    vMax=5;	  
  }//Fine costruttore

SimpleModelParser::~SimpleModelParser(){}
	
	
float SimpleModelParser::getVMax()
   {
     if(vMax<0)  
	 return 0;
	else
     return vMax;		
   }//Fine getVMax	
   
float SimpleModelParser::getVMin()
   {
	if(vMin<0)  
	 return 0;
	else
     return vMin;		
   }//Fine getVMax  

float SimpleModelParser::getPauseTime()
   {
     if(pauseTime<0)  
	 return 0;
	else
     return pauseTime;		
   }//Fine getInterval    
   
	
	
void SimpleModelParser::OnStartElement(const XML_Char* name,const XML_Char** attrs)
  {
    //Richiamo metodo del padre
    ModelParser::OnStartElement(name,attrs);	  
	  
    std::string str=xmlCharVectToString(name);	     
    	  
     //Intervallo di tempo 
    if(str=="pauseTime")
	 { actTag=SimpleModelParser::PAUSETIME; }
    
    //Velocita' minima 
    if(str=="vMin")
	 { actTag=SimpleModelParser::VMIN; }

    //Velocita' massima 
    if(str=="vMax")
	 { actTag=SimpleModelParser::VMAX; } 		  
    
  }//Fine OnStartElement	
  
  
void SimpleModelParser::OnCharacterData(const XML_Char* data, int len)
  {
    //Richiamo metodo del padre
    ModelParser::OnCharacterData(data,len);
	  
    std::string str=xmlCharVectToString(data,len);	
  
     //Intervallo
    if(actTag==SimpleModelParser::PAUSETIME) 
	{ 
          pauseTime=stringToFloat(str);
          actTag=-1;
        }

    //Velocita' minima
    if(actTag==SimpleModelParser::VMIN) 
	{
         vMin=stringToFloat(str);
         actTag=-1;
       }
   
    //Velocità massima
    if(actTag==SimpleModelParser::VMAX) 
	{ 
         vMax=stringToFloat(str);
         actTag=-1;
     }
 }//Fine  OnCharacterData
  
  
void SimpleModelParser::OnEndElement(const XML_Char* name){}  
	
